source/game/field/obj/ObjectRock.hh

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "game/field/obj/ObjectCollidable.hh"
+
+namespace Field {
+
+class ObjectRock : public ObjectCollidable {
+public:
+    ObjectRock(const System::MapdataGeoObj &params);
+    ~ObjectRock() override;
+
+    void init() override;
+    void calc() override;
+
+    /// @addr{0x8077037C}
+    [[nodiscard]] u32 loadFlags() const override {
+        return 1;
+    }
+
+    /// @addr{0x8077036C}
+    [[nodiscard]] const EGG::Vector3f &getCollisionTranslation() const override {
+        return m_colTranslate;
+    }
+
+    [[nodiscard]] Kart::Reaction onCollision(Kart::KartObject *kartObj,
+            Kart::Reaction reactionOnKart, Kart::Reaction reactionOnObj,
+            EGG::Vector3f &hitDepth) override;
+
+    void calcTangible();
+    void calcIntangible();
+    void calcTangibleSub();
+
+    void checkSphereFull();
+    void breakRock();
```

Comment:
Can you move all of these into the `private` section? In between the `enum class` declaration and the member variable declarations. We don't need to expose these functions outside of the class.